python/tchannel/tornado/dispatch.py

```diff
@@ -38,3 +135,84 @@ def _call_endpoint(self, endpoint, request, response):
         ioloop.IOLoop.current().add_future(future, lambda f: f.exception())
 
         return future
+
+
+class Request(object):
+    """Represents an incoming request to an endpoint.
+
+    Request class is used to represent the CallRequestMessage at User's level.
+    This is going to hide the protocol level message information.
+    """
+
+    # TODO decide which elements inside "message" object to expose to user.
+    def __init__(
+        self,
+        id=None,
+        flags=FlagsType.none,
+        ttl=10,
+        tracing=None,
+        service=None,
+        headers=None,
+        checksum=None,
+        argstreams=None
+    ):
+        self.flags = flags
+        self.ttl = ttl
+        self.service = service
+        # argstreams is a list of InMemArgStream/PipeArgStream objects
+        self.argstreams = argstreams
+        self.id = id
+        self.headers = headers
+        self.state = StreamState.init
+        self.endpoint = ""
+
+    @property
+    def arg_scheme(self):
+        return self.headers.get('as', None)
```

Comment:
The default value for headers is None. This will throw an exception. Check that it's non None.